README.md
## Cooldowns by Yarko

A World of Warcraft add-on that displays numeric cooldown counters on buttons and other icons with cooldowns.

Configure colors, fonts, and other options through the Cooldowns interface options panel.

## Using a different font in Cooldowns

1. If the font file you want to use is already located somewhere in the World of Warcraft subfolders, go to step 5. If you want to use a font file that does not already exist in the WoW file system, go to step 2.

2. Shut down WoW. When copying a new file into the WoW subfolders, WoW must be restarted or the program won't see it.

3. Copy the font file to a folder somewhere in the WoW file system. For example, you can create a new folder called Fonts in the World of Warcraft Folder and copy the file there, or you can copy the file into any of the Interface\Addons folders. Where you put it is completely up to you. Just be sure to put it someplace that makes sense and where you won't lose it. Sticking fonts into the screenshots folder doesn't make a lot of sense, for example.

4. Start WoW.

5. Go into the Cooldowns options window by going to the main menu, clicking on the Interface button, clicking on the Addons tab, and clicking on the Cooldowns menu option in the list.

6. Enter the location of the font file. This would be the path to the folder that contains the font file relative to the main WoW folder. Upper and lower case don't matter here. WoW file folder locations are always relative to the main WoW folder. For example, if you created a Fonts folder off of the main WoW folder, then enter "fonts" into the Font Location field. This is the default value since WoW uses a "hidden" folder called Fonts to store the default fonts that come with the game. The addon Recount comes with a font called ABF.ttf. Relative to the main WoW folder, that file is located in Interface\Addons\Recount\Fonts. So, to use this font in Cooldowns, you would enter "interface\addons\recount\fonts" in the location field.

7. Enter the Font File Name. Again, upper and lower case don't matter. For the Recount font file that was mentioned in the last step, you would enter "abf.ttf" as the Font File Name.

8. Set the font height. You might want to adjust this value to make some fonts more readable. Play around with it as needed.

## WeakAuras Compatibility

To disable the cooldown count for a WeakAuras icon aura, call `YarkoCooldowns.WeakAurasIcon_DisableCooldownCount(aura_env.region)` from the On Init action of the aura.


CHANGELOG.md
## 4.6
- Allow disabling cooldown count for WeakAuras icons

## 4.5.3
- Fix Profession Trait Progress Bars not being ignored

## 4.5.2
- Ignore Profession Trait Progress Bars

## 4.5.1
- Fix some default UI AddOns not being ignored

## 4.5
- Ignore Faction Renown and Profession Specialization progress bars
- Update to new Settings system
- Bump TOC Interface version to 10.0.0

## 4.4.1
- Fix incorrect variable name

## 4.4
- Ignore default UI elements that use Cooldown widgets but aren't cooldowns
- Bump TOC Interface version to 9.0.2
- Add TOC properties, changelog, .pkgmeta and GitHub Actions workflow for the BigWigs packager script
    - https://github.com/BigWigsMods/packager/wiki/GitHub-Actions-workflow

## 4.3.2
- Update to 7.1

## 4.3.1
- Updated toc to 50400

## 4.3
- Changed action bar button handling to use cooldown event

## 4.2.4
- TOC to 50300

## 3.0.2
- Fixed problem with counter scaling when running with Bartender4.
- Removed support for ButtonFacade since ButtonFacade is now Masque.

## 2.1
- Added options to disable cooldown number flashing and how many seconds to flash
- Removed upper limit for setting how many seconds to show seconds.
- Fixed problems with cooldown text sizing

Cooldowns.lua
YarkoCooldowns = {}

YarkoCooldowns.DefaultMainColor = NORMAL_FONT_COLOR
YarkoCooldowns.DefaultFlash = "Y"
YarkoCooldowns.DefaultFlashSeconds = 3
YarkoCooldowns.DefaultAlternate = 1
YarkoCooldowns.DefaultFlashColor = { r = 1.0, g = 0, b = 0 }
YarkoCooldowns.DefaultFontLocation = "Fonts"
YarkoCooldowns.DefaultFontFile = "FRIZQT__.TTF"
YarkoCooldowns.DefaultFontHeight = 26
YarkoCooldowns.DefaultShadow = "N"
YarkoCooldowns.DefaultOutline = 3
YarkoCooldowns.DefaultTenths = "N"
YarkoCooldowns.DefaultBelowTwo = "N"
YarkoCooldowns.DefaultSeconds = 60
YarkoCooldowns.DefaultMinimum = 1.5
YarkoCooldowns.DefaultSize = 27

YarkoCooldowns.CooldownFrames = {}


local ActiveCounters = {}

local TimeSinceLastUpdate = 1
local UpdateInterval = .1

local TimeSinceLastFlash = 0
local FlashInterval = .3
local FlashFlag = false

local Scales = {
	[1] = 1,
	[2] = .85,
	[3] = .65,
	[4] = .5
}

local SpecialAddon

local OutlineList = { nil, "OUTLINE", "THICKOUTLINE" }

function YarkoCooldowns.Test()
	for k, v in pairs(ActiveCounters) do
		print(k, v, v:GetName())
	end
end

function YarkoCooldowns.OnLoad(self)
	self:RegisterEvent("ADDON_LOADED")
	self:RegisterEvent("PLAYER_LOGIN")
	self:RegisterEvent("ACTIONBAR_UPDATE_COOLDOWN")
end

function YarkoCooldowns.OnEvent(_, event, ...)
	if event == "ADDON_LOADED" then
		YarkoCooldowns.DisableCooldownsForDefaultUIAndAddOns()

		return
	end

	if event == "PLAYER_LOGIN" then
		SpecialAddon = Bartender4

		YarkoCooldowns.OptionsSetup()

		-- Does not work with addons that call SetCooldown() directly
		--hooksecurefunc("CooldownFrame_SetTimer", YarkoCooldowns.CooldownSetTimer)

		-- So hook directly to the cooldown widget function
		hooksecurefunc(getmetatable(ActionButton1Cooldown).__index, "SetCooldown", YarkoCooldowns.SetCooldown)

		YarkoCooldowns.DisableCooldownsForDefaultUIAndAddOns()

		return
	end

	if event == "ACTIONBAR_UPDATE_COOLDOWN" then
		for _, button in ipairs(ActionBarButtonEventsFrame.frames) do
			if button.cooldown:IsVisible() then
				local start, duration

				if button.spellID then
					start, duration = GetSpellCooldown(button.spellID)
				else
					start, duration = GetActionCooldown(button.action)
				end

				if start > 0 then
					YarkoCooldowns.StartCooldown(button.cooldown, start, duration, (duration > 0 and duration < 1 and 0) or 1)
				end
			end
		end
	end
end

function YarkoCooldowns.SetCooldown(self, start, duration, modRate)
	if not self.noCooldownCount and start > 0 then
		YarkoCooldowns.StartCooldown(self, start, duration, (duration > 0 and duration < 1 and 0) or 1)
	end
end

function YarkoCooldowns.StartCooldown(self, start, duration, enable)
	if (start > 0 and enable > 0 and duration > 1.5
		and (start + duration - GetTime() - .01 > YarkoCooldowns_SavedVars.Minimum or ActiveCounters[self])) then
		local frame = YarkoCooldowns.CooldownFrames[self]

		if not frame then
			local parent = self:GetParent()
			local parentName = parent:GetName()
			local name, bt4Frame

			if parentName then
				name = "YarkoCooldowns" .. parentName
				bt4Frame = SpecialAddon and parentName:sub(1, 3) == "BT4"
			end

			-- For Bartender, don't parent counter to cooldown so that button can be made trasparent and cooldown still visible
			frame = CreateFrame("Frame", name, not bt4Frame and self or nil, "YarkoCooldowns_CounterTemplate")
			frame:SetPoint("CENTER", parent, "CENTER")

			if bt4Frame then
				frame:SetFrameStrata(parent:GetFrameStrata())
			end

			frame:SetFrameLevel(self:GetFrameLevel() + 5)

			self:HookScript("OnShow", function(self) YarkoCooldowns.CooldownFrames[self]:Show() end)
			self:HookScript("OnHide", function(self) YarkoCooldowns.CooldownFrames[self]:Hide() end)

			YarkoCooldowns.UpdateFont(frame)
			frame:Show()

			frame.cooldown = self
			frame.bt4Frame = bt4Frame
			frame.filtername = YarkoCooldowns.GetParentFrame(parent:GetParent())

			YarkoCooldowns.CooldownFrames[self] = frame
		end

		frame.start = start
		frame.duration = duration
		frame.enable = enable
		frame.flag = false

		ActiveCounters[self] = frame
	else
		if ActiveCounters[self] then
			ActiveCounters[self].text1:SetText("")
			ActiveCounters[self].text2:SetText("")
			ActiveCounters[self] = nil
		end
	end
end

function YarkoCooldowns.OnUpdate(_, elapsed)
	TimeSinceLastFlash = TimeSinceLastFlash + elapsed

	if TimeSinceLastFlash > FlashInterval then
		FlashFlag = not FlashFlag
		TimeSinceLastFlash = 0
	end

	TimeSinceLastUpdate = TimeSinceLastUpdate + elapsed

	if TimeSinceLastUpdate > UpdateInterval then
		for cooldown, frame in pairs(ActiveCounters) do
			local timeLeft = GetTime() > frame.start and frame.start + frame.duration - GetTime() or 0
			local counterText1 = frame.text1
			local counterText2 = frame.text2

			if timeLeft > 0 and frame.enable > 0 then
				local parent = cooldown:GetParent()
				local parentframe = frame.filtername

				if parentframe and not YarkoCooldowns_SavedVars.ParentFrames[parentframe] then
					YarkoCooldowns_SavedVars.ParentFrames[parentframe] = "Y"
				end

				if YarkoCooldowns.Round(parent:GetWidth()) * parent:GetEffectiveScale() / UIParent:GetScale() >=
					YarkoCooldowns_SavedVars.Size
					and not cooldown.noCooldownCount
					and cooldown:IsVisible() and parent:GetEffectiveAlpha() > 0
					and (not parentframe or YarkoCooldowns_SavedVars.ParentFrames[parentframe] ~= "N")
				then
					local line1, line2, length

					if timeLeft > 3600 then
						line1 = YarkoCooldowns_SavedVars.Tenths == "Y"
							and YarkoCooldowns.TrimZeros(("%.1f"):format((timeLeft + 180) / 3600))
							or ("%d"):format(ceil(timeLeft / 3600))
						line2 = "h"
					elseif timeLeft > YarkoCooldowns_SavedVars.Seconds then
						line1 = ((YarkoCooldowns_SavedVars.Tenths == "Y"
							and YarkoCooldowns.TrimZeros(("%.1f"):format((timeLeft + 3) / 60)))
							or ("%d"):format(ceil(timeLeft / 60)))
						line2 = "m"
					else
						line1 = ((timeLeft <= 2 and YarkoCooldowns_SavedVars.BelowTwo == "Y"
							and ("%.1f"):format(timeLeft + 0.05))
							or ("%d"):format(ceil(timeLeft)))
						line2 = ""
					end

					length = #line1

					if #line2 > 0 and length == 1 then
						line1 = line1 .. line2
						line2 = ""
						length = length + 1
					end

					if line1 ~= counterText1:GetText() or line2 ~= counterText2:GetText() then
						if length > 4 then
							length = 4
						end

						frame:SetScale(
							Scales[length] * parent:GetWidth() / ActionButton1:GetWidth()
							* (frame.bt4Frame and parent:GetEffectiveScale() or 1)
							* (#line2 > 0 and length == 2 and 0.8 or 1)
							* (timeLeft <= 2 and YarkoCooldowns_SavedVars.BelowTwo == "Y" and 0.95 or 1)
						)

						counterText1:ClearAllPoints()

						local x = length == 4 and 4 or 2

						if #line2 < 1 then
							counterText1:SetPoint("CENTER", frame, "CENTER", x, -3)
						else
							counterText1:SetPoint("BOTTOM", frame, "TOP", x, -23)
						end

						counterText1:SetText(line1)
						counterText2:SetText(line2)
					end

					if timeLeft <= YarkoCooldowns_SavedVars.FlashSeconds and YarkoCooldowns_SavedVars.Flash == "Y" then
						if YarkoCooldowns_SavedVars.Alternate == 1 then
							frame.flag = FlashFlag
						else
							if not frame.flag then
								frame.flag = true
								frame.oldflag = false
							end
						end
					else
						if frame.flag then
							frame.flag = false
							frame.oldflag = true
						end
					end

					if frame.flag ~= frame.oldflag then
						YarkoCooldowns.SetCounterColor(counterText1, frame.flag)
						YarkoCooldowns.SetCounterColor(counterText2, frame.flag)
						frame.oldflag = frame.flag
					end
				else
					counterText1:SetText("")
					counterText2:SetText("")
				end
			else
				counterText1:SetText("")
				counterText2:SetText("")
				ActiveCounters[cooldown] = nil
			end
		end

		TimeSinceLastUpdate = 0
	end
end

function YarkoCooldowns.TrimZeros(inStr)
	local outStr = ""
	local str = ""

	for i = 1, #inStr do
		if inStr:sub(i, i) ~= "0" then
			str = inStr:sub(i)
			break
		end
	end

	for i = #str, 1, -1 do
		if str:sub(i, i) == "." then
			outStr = str:sub(1, i - 1)
			break
		end

		if str:sub(i, i) ~= "0" then
			outStr = str:sub(1, i)
			break
		end
	end

	return outStr
end

function YarkoCooldowns.UpdateFont(cooldownFrame)
	local counterText1 = cooldownFrame.text1
	local counterText2 = cooldownFrame.text2

	if YarkoCooldowns_SavedVars.Shadow == "Y" then
		counterText1:SetShadowOffset(1, -1)
		counterText2:SetShadowOffset(1, -1)
	else
		counterText1:SetShadowOffset(0, 0)
		counterText2:SetShadowOffset(0, 0)
	end

	counterText1:SetFont(
		YarkoCooldowns_SavedVars.FontLocation .. "\\" .. YarkoCooldowns_SavedVars.FontFile,
		YarkoCooldowns_SavedVars.FontHeightX,
		OutlineList[YarkoCooldowns_SavedVars.Outline]
	)

	counterText2:SetFont(
		YarkoCooldowns_SavedVars.FontLocation .. "\\" .. YarkoCooldowns_SavedVars.FontFile,
		YarkoCooldowns_SavedVars.FontHeightX,
		OutlineList[YarkoCooldowns_SavedVars.Outline]
	)

	YarkoCooldowns.SetCounterColor(counterText1, cooldownFrame.flag or false)
	YarkoCooldowns.SetCounterColor(counterText2, cooldownFrame.flag or false)
end

function YarkoCooldowns.SetCounterColor(counterText, flag)
	if not flag then
		counterText:SetTextColor(
			YarkoCooldowns_SavedVars.MainColor.r,
			YarkoCooldowns_SavedVars.MainColor.g,
			YarkoCooldowns_SavedVars.MainColor.b
		)
	else
		counterText:SetTextColor(
			YarkoCooldowns_SavedVars.FlashColor.r,
			YarkoCooldowns_SavedVars.FlashColor.g,
			YarkoCooldowns_SavedVars.FlashColor.b
		)
	end
end

function YarkoCooldowns.GetParentFrame(frame)
	local name

	repeat
		name = frame:GetName()
		frame = frame:GetParent()
	until name or not frame

	return name == "UIParent" and nil or name
end

function YarkoCooldowns.Round(num)
	return math.floor(num + 0.5)
end


Localization.de.lua
-- Thanks to zreptil and Darkwing151 on Curse

-- German
if GetLocale() == "deDE" then
	YARKOCOOLDOWNS_CONFIG_MAINCOLOR = "Cooldown Zahlen Farbe"
	YARKOCOOLDOWNS_CONFIG_FLASH = "Benutze alternative Cooldownfarbe ab"
	YARKOCOOLDOWNS_CONFIG_FLASH2 = "Sekunden"
	YARKOCOOLDOWNS_CONFIG_ALTERNATE = "Zeige alternative Farbe als"
	YARKOCOOLDOWNS_CONFIG_ALTFLASH = "blinkend"
	YARKOCOOLDOWNS_CONFIG_ALTSOLID = "solide"
	YARKOCOOLDOWNS_CONFIG_FLASHCOLOR = "Cooldown blinkende Farbe"
	YARKOCOOLDOWNS_CONFIG_FONTLOCATION = "Font Verzeichnis"
	YARKOCOOLDOWNS_CONFIG_FONTFILE = "Font Dateiname"
	YARKOCOOLDOWNS_CONFIG_FONTHEIGHT = "Font H\195\182he"
	YARKOCOOLDOWNS_CONFIG_SHADOW = "Text Schatten"
	YARKOCOOLDOWNS_CONFIG_OUTLINE = "Text Umriss"
	YARKOCOOLDOWNS_CONFIG_NONE = "Keiner"
	YARKOCOOLDOWNS_CONFIG_NORMAL = "D\195\188nn"
	YARKOCOOLDOWNS_CONFIG_THICK = "Dick"
	YARKOCOOLDOWNS_CONFIG_TENTHS = "Zeige Zehntel bei Minuten und Stunden"
	YARKOCOOLDOWNS_CONFIG_BELOWTWO = "Zeige Zehntel unter zwei Sekunden"
	YARKOCOOLDOWNS_CONFIG_SECONDS = "Zeige Sekunden bei"
	YARKOCOOLDOWNS_CONFIG_SECONDS2 = "Sekunden"
	YARKOCOOLDOWNS_CONFIG_MINIMUM = "Zeige Cooldowntext nur wenn der Cooldown l\195\164nger als"
	YARKOCOOLDOWNS_CONFIG_MINIMUM2 = "Sekunden ist"
	YARKOCOOLDOWNS_CONFIG_SIZE = "Zeige Cooldowntext nur wenn das Icon"
	YARKOCOOLDOWNS_CONFIG_SIZE2 = "Pixel oder gr\195\182\195\159er ist"
	YARKOCOOLDOWNS_CONFIG_TAB1 = "Allgemeine Einstellungen"
	YARKOCOOLDOWNS_CONFIG_TAB2 = "Filter"
	YARKOCOOLDOWNS_CONFIG_LIST = "Erkannte Cooldown Frames"
	YARKOCOOLDOWNS_CONFIG_REMOVE = "Entfernen: Alle"
	YARKOCOOLDOWNS_CONFIG_UNSELECTED = "Entfernen: Unmarkiert"

	YARKOCOOLDOWNS_CONFIG_FONTLOCATION_INFO = "Ordner  innerhalb des WoW Dateisystems, wo die Fonts plaziert sind. Benutze \"Fonts\" "
		..
		"f\195\188r Standard In-Game Fonts wie FRIZQT__.TTF. Um eine Datei in einem Addon Unterordner auszuw\195\164hlen, gib an: "
		..
		"\"Interface\\Addons\\<Name des Addons>\\<Unterordner>\"."

	YARKOCOOLDOWNS_CONFIG_FONTFILE_INFO = "Name der Font Datei, wie \"FRIZQT__.TTF\" oder \"ARIALN.TTF\""

	YARKOCOOLDOWNS_CONFIG_LIST_INFO = "Diese Liste zeigt erkannte Cooldown Frames auf denen ein Cooldowntext platziert wird, durch "
		..
		"demarkieren wird der Cooldowntext bei diesem Frame versteckt. Diese Liste erweitert sich automatisch.\n\nBenutzen der unteren "
		..
		"Schaltfl\195\164chen entfernt Eintr\195\164ge aus dieser Liste, dies ist zum Beispiel sinnvoll um alte nichtmehr benutzte Frames zu entfernen. "
		..
		"Entfernte Eintr\195\164ge werden wieder mit Cooldowntext dargestellt."

	YARKOCOOLDOWNS_CONFIG_SIZE_INFO = "Standard action button: 36\nStandard player buff: 30\nStandard target frame debuff: 17"
end


Cooldowns_Compat.lua
local ProcessedAddOns = {
	Blizzard_GarrisonUI = false,
	Blizzard_MajorFactions = false,
	Blizzard_Professions = false,
	Blizzard_PVPUI = false,
	Blizzard_UIWidgets = false,
	WeakAuras = false,
}

local function CheckAddOn(addOnName)
	if IsAddOnLoaded(addOnName) and not ProcessedAddOns[addOnName] then
		ProcessedAddOns[addOnName] = true
		return true
	end

	return false
end

function YarkoCooldowns.DisableCooldownsForDefaultUIAndAddOns()
	if CheckAddOn("Blizzard_GarrisonUI") then
		-- Ignore the follower experience bar for Covenant Mission rewards.
		-- The follower frames are created on-demand from a frame pool, so we hook the mixin functions before the frames are created.

		local function AdventuresRewardsFollowerMixin_DisableCooldownCount(self)
			self.noCooldownCount = true
		end

		hooksecurefunc(AdventuresRewardsFollowerMixin, "SetFollowerInfo", AdventuresRewardsFollowerMixin_DisableCooldownCount)
		hooksecurefunc(AdventuresRewardsFollowerMixin, "UpdateExperience", AdventuresRewardsFollowerMixin_DisableCooldownCount)
	end

	if CheckAddOn("Blizzard_MajorFactions") then
		-- Ignore the Renown progress bars.
		-- MajorFactionRenownFrame is created statically, so we apply the changes directly to the RenownProgressBar frame.
		-- MajorFactionButtonTemplate buttons are created on-demand, so we hook the mixin function before the frames are created.

		MajorFactionRenownFrame.HeaderFrame.RenownProgressBar.noCooldownCount = true

		local function MajorFactionButtonMixin_Init(self)
			self.UnlockedState.RenownProgressBar.noCooldownCount = true
		end

		hooksecurefunc(MajorFactionButtonMixin, "Init", MajorFactionButtonMixin_Init)
	end

	if CheckAddOn("Blizzard_Professions") then
		-- Ignore the Professions Specialization progress bars.
		-- ProfessionsFrame is created statically, so we apply the changes directly to the Path ProgressBar frame.
		-- Profession Trait Buttons are created on-demand, so we listen to the TalentButtonAcquired callback.

		ProfessionsFrame.SpecPage.DetailedView.Path.ProgressBar.noCooldownCount = true

		ProfessionsFrame.SpecPage:RegisterCallback(TalentFrameBaseMixin.Event.TalentButtonAcquired,
			function(_, talentButton)
				talentButton.ProgressBar.noCooldownCount = true
			end)

		ProfessionsFrame.SpecPage:RegisterCallback(TalentFrameBaseMixin.Event.TalentButtonReleased,
			function(_, talentButton)
				talentButton.ProgressBar.noCooldownCount = nil
			end)

		for talentButton in ProfessionsFrame.SpecPage:EnumerateAllTalentButtons() do
			talentButton.ProgressBar.noCooldownCount = true
		end
	end

	if CheckAddOn("Blizzard_PVPUI") then
		-- Ignore the Honor Level display.
		-- PVPUIFrame is created statically, so we apply the changes directly to the HonorLevelDisplay frame.

		PVPQueueFrame.HonorInset.CasualPanel.HonorLevelDisplay.noCooldownCount = true
	end

	if not ProcessedAddOns.Blizzard_UIWidgets then
		-- Ignore the UIWidgetBaseCircularStatusBarTemplate used by UIWidgetTemplateDiscreteProgressSteps for The Eye of the Jailer in the Maw.
		-- Ignore the UIWidgetBaseControlZoneTemplate used by UIWidgetTemplateCaptureZone/UIWidgetTemplateZoneControl for what seems to be an event in the Sophia's Overture subzone of Bastion.
		-- UIWidgets are created on-demand from frame pools, so we hook the mixin functions before the frames are created.

		ProcessedAddOns.Blizzard_UIWidgets = true

		local function UIWidgetBaseCircularStatusBar_DisableCooldownCount(self)
			self.noCooldownCount = true
		end

		local function UIWidgetBaseControlZone_DisableCooldownCount(self)
			self.UncapturedSection.noCooldownCount = true
			self.Progress.noCooldownCount = true
		end

		hooksecurefunc(UIWidgetBaseCircularStatusBarTemplateMixin, "Setup", UIWidgetBaseCircularStatusBar_DisableCooldownCount)
		hooksecurefunc(UIWidgetBaseControlZoneTemplateMixin, "Setup", UIWidgetBaseControlZone_DisableCooldownCount)
	end

	if CheckAddOn("WeakAuras") then
		-- Provide function to ignore WeakAuras icon cooldowns
		function YarkoCooldowns.WeakAurasIcon_DisableCooldownCount(region)
			local cooldown = region.cooldown
			if cooldown then
				cooldown.noCooldownCount = true
			end
		end
	end
end


Localization.lua
YARKOCOOLDOWNS_TITLE = "Cooldowns"

YARKOCOOLDOWNS_CONFIG_MAINCOLOR = "Cooldown number color"
YARKOCOOLDOWNS_CONFIG_FLASH = "Use cooldown number alternate color at"
YARKOCOOLDOWNS_CONFIG_FLASH2 = "seconds"
YARKOCOOLDOWNS_CONFIG_ALTERNATE = "Show alternate color as"
YARKOCOOLDOWNS_CONFIG_ALTFLASH = "Flashing"
YARKOCOOLDOWNS_CONFIG_ALTSOLID = "Solid"
YARKOCOOLDOWNS_CONFIG_FLASHCOLOR = "Cooldown alternate color"
YARKOCOOLDOWNS_CONFIG_FONTLOCATION = "Font Location"
YARKOCOOLDOWNS_CONFIG_FONTFILE = "Font File Name"
YARKOCOOLDOWNS_CONFIG_FONTHEIGHT = "Font Height"
YARKOCOOLDOWNS_CONFIG_SHADOW = "Text Shadow"
YARKOCOOLDOWNS_CONFIG_OUTLINE = "Text Outline"
YARKOCOOLDOWNS_CONFIG_NONE = "None"
YARKOCOOLDOWNS_CONFIG_NORMAL = "Thin"
YARKOCOOLDOWNS_CONFIG_THICK = "Thick"
YARKOCOOLDOWNS_CONFIG_TENTHS = "Show tenths for minutes and hours"
YARKOCOOLDOWNS_CONFIG_BELOWTWO = "Show tenths below two seconds"
YARKOCOOLDOWNS_CONFIG_SECONDS = "Show seconds at"
YARKOCOOLDOWNS_CONFIG_SECONDS2 = "seconds"
YARKOCOOLDOWNS_CONFIG_MINIMUM = "Show cooldown text only when cooldown longer than"
YARKOCOOLDOWNS_CONFIG_MINIMUM2 = "seconds"
YARKOCOOLDOWNS_CONFIG_SIZE = "Show cooldown text only when cooldown icon width"
YARKOCOOLDOWNS_CONFIG_SIZE2 = "pixels or greater"
YARKOCOOLDOWNS_CONFIG_TAB1 = "General Settings"
YARKOCOOLDOWNS_CONFIG_TAB2 = "Filtering"
YARKOCOOLDOWNS_CONFIG_LIST = "Recognized cooldown frames"
YARKOCOOLDOWNS_CONFIG_REMOVE = "Remove All"
YARKOCOOLDOWNS_CONFIG_UNSELECTED = "Remove Unselected"


YARKOCOOLDOWNS_CONFIG_FONTLOCATION_INFO = "Directory within the WoW file system where font is located. Use \"Fonts\" "
	.. "for standard in-game fonts such as FRIZQT__.TTF. To indicate a file in an Addons subfolder, use "
	.. "\"Interface\\Addons\\<add-on name>\\<subfolder>\"."

YARKOCOOLDOWNS_CONFIG_FONTFILE_INFO = "Name of the font file, such as \"FRIZQT__.TTF\" or \"ARIALN.TTF\""

YARKOCOOLDOWNS_CONFIG_LIST_INFO = "This list contains the names of all frames that Cooldowns currently recognizes for "
	..
	"placing cooldown text. More entries are added as new cooldowns are used. Uncheck list entries to keep Cooldowns from placing text on objects in those frames.\n\n"
	..
	"Using the buttons below the list removes entries from the list essentially forcing Cooldowns to rerecognize frames for the "
	..
	"placement of cooldown text. These buttons are useful to make Cooldowns forget about frames you may no longer be "
	..
	"using such as when switching or abandoning addons. Removing entries means that cooldown text will appear on those frames again."

YARKOCOOLDOWNS_CONFIG_SIZE_INFO = "Standard action button: 36\nStandard player buff: 30\nStandard target frame debuff: 17"
